### ISSUE TITLE
side effects in @cmd operations

### ISSUE THREAD

**Issue by berceanu** [glotzerlab/signac-flow#70]
```python
from flow import FlowProject, cmd
import signac

pr = signac.init_project('issue_163')
job = pr.open_job({}).init()

def _shell_command(job, envvar):
    command = f"echo ${envvar} >> var.txt"
    print(f"Executing `{command}`...")
    return command


@FlowProject.operation
@cmd
@FlowProject.post.isfile('var.txt')
def run_shell_command(job):
    return _shell_command(job, envvar='PATH')

if __name__ == '__main__':
    FlowProject().main()
```

**Expected console output:**
```console
Executing `echo $PATH >> var.txt`...
```

**Actual console output:**
```console
Executing `echo $PATH >> var.txt`...
Executing `echo $PATH >> var.txt`...
```

A use-case for this would be logging the executed `@cmd` to a logfile upon execution.
The function `_shell_command` is executed, even though `command` is not actually run.

**Comment by csadorf** [comment]
Thank you for reporting! While `@cmd` operation-functions *should* technically not have any side-effects, it would probably a good idea if we would avoid executing them regardless.

**Comment by bdice** [comment]
Resolved by #396.
